angular-ngrx-scss/src/app/state/auth/auth.effects.ts

```diff
@@ -49,9 +70,40 @@ export class AuthEffects {
     );
   });
 
+  /**
+   * Gets authenticated user's name, photo, and email
+   */
+  fetchAuthUserData$ = createEffect(() => {
+    return this.actions$.pipe(
+      ofType(saveUserTokenSuccess, userTokenExists),
+      switchMap(() =>
+        this.store.select(selectAuthUserName).pipe(
+          skipWhile((name) => name.length > 0),
+          mergeMap(() =>
```

Comment:
do we actually want to use `mergeMap` here? If for some reason the selector emits multiple times we would end up with multiple requests happening at once and a race condition. Maybe this is the rare place to use `exhaustMap`?